### PR TITLE
Use buildkit while building images

### PR DESCRIPTION
**Pull request by wendigo**
Buildkit is faster and does caching better when building images.
In the near future, it will become a default docker build tooling.

### CODE REVIEW

**Comment by findepi** [comment]
Can you add some PR and commit description? Especially -- why we would want to do the change.

**Review by kokosing — APPROVED** [review]

**Review by hashhar — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -190,9 +190,9 @@ $(UNLABELLED_TAGS): %@unlabelled: %/Dockerfile %@latest
 #
 $(LATEST_TAGS): %@latest: %/Dockerfile %-parent-check
 	@echo
-	@echo "Building [$@] image"
+	@echo "Building [$@] image using buildkit"
 	@echo
-	cd $* && time $(SHELL) -c "( tar -czh . | DOCKER_BUILDKIT=0 docker build ${BUILD_ARGS} $(DBFLAGS_$*) -t $(call docker-tag,$@) --label $(LABEL) - )"
```
Comment (findepi) [inline Makefile:195]: do you know why we had `DOCKER_BUILDKIT=0` here?

File: Makefile [hunk 2]
```diff
@@ -190,9 +190,9 @@ $(UNLABELLED_TAGS): %@unlabelled: %/Dockerfile %@latest
...
+	cd $* && time $(SHELL) -c "( docker buildx build --squash --compress --add-host hadoop-master:127.0.0.2 ${BUILD_ARGS} $(DBFLAGS_$*) -t $(call docker-tag,$@) --label $(LABEL) . )"
```
Comment (findepi) [inline Makefile:195]: I don't think we should apply `hadoop-master` to every image.
some are not hadoop-related, an image should opt-in into that

Comment (wendigo) [reply Makefile:195]: That's a build time only. It won't be set when image is ran

[the remark this answers is not in the document]
Comment (wendigo) [reply Makefile:195]: To satisfy requirement of different setup scripts. It must be different from localhost but still point to loopback

Comment (wendigo) [reply Makefile:195]: You cannot modify hosts from within an image as it's read only fs with buildkit

Comment (electrum) [inline Makefile:195]: Why `--squash`? This can make caching less effective if only the later layers change.

Comment (wendigo) [reply Makefile:195]: I'll remove squash

**Pull request merged**